src/dezerialize.ts

```diff
@@ -69,96 +69,148 @@ type OmitKey<T, K> = DistributiveOmit<T, keyof K>;
-  ? z.ZodUnknown
-  : T extends SzNever
-  ? z.ZodNever
-  : T extends SzVoid
-  ? z.ZodVoid
-  : T extends SzLiteral<infer Value>
-  ? z.ZodLiteral<Value> // List Collections
-  : T extends SzTuple<infer _Items>
-  ? z.ZodTuple<any> //DezerializeArray<Items>>
-  : T extends SzSet<infer Value>
-  ? z.ZodSet<Dezerialize<Value>>
-  : T extends SzArray<infer Element>
-  ? z.ZodArray<Dezerialize<Element>> // Key/Value Collections
-  : T extends SzObject<infer Properties>
-  ? z.ZodObject<{
-      [Property in keyof Properties]: Dezerialize<Properties[Property]>;
-    }>
-  : T extends SzRecord<infer Key, infer Value>
-  ? z.ZodRecord<Dezerialize<Key>, Dezerialize<Value>>
-  : T extends SzMap<infer Key, infer Value>
-  ? z.ZodMap<Dezerialize<Key>, Dezerialize<Value>> // Enum
-  : T extends SzEnum<infer Values>
-  ? z.ZodEnum<Values> // Union/Intersection
-  : T extends SzUnion<infer _Options>
-  ? z.ZodUnion<any>
-  : T extends SzDiscriminatedUnion<infer Discriminator, infer _Options>
-  ? z.ZodDiscriminatedUnion<Discriminator, any>
-  : T extends SzIntersection<infer L, infer R>
-  ? z.ZodIntersection<Dezerialize<L>, Dezerialize<R>> // Specials
-  : T extends SzFunction<infer Args, infer Return>
-  ? z.ZodFunction<Dezerialize<Args>, Dezerialize<Return>>
-  : T extends SzPromise<infer Value>
-  ? z.ZodPromise<Dezerialize<Value>>
-  : T extends SzEffect<infer Value>
-  ? z.ZodEffects<Dezerialize<Value>>
-  : T extends SzCatch<infer Value>
-  ? z.ZodCatch<Dezerialize<Value>>
-  : T extends SzNativeEnum<infer Value>
-  ? z.ZodNativeEnum<Value>
-  : unknown;
+    : T extends SzNullable
+      ? Dezerialize<OmitKey<T, SzNullable>> extends infer I
+        ? I extends ZodTypes
+          ? z.ZodNullable<I>
+          : never
+        : never
+      : T extends SzReadonly
+        ? Dezerialize<OmitKey<T, SzReadonly>> extends infer I
+          ? I extends ZodTypes
+            ? z.ZodReadonly<I>
+            : never
+          : never
+        : T extends SzDefault<any>
+          ? Dezerialize<OmitKey<T, SzDefault<any>>> extends infer I
+            ? I extends ZodTypes
+              ? z.ZodDefault<I>
+              : never
+            : never // Primitives
+          : T extends SzString
+            ? z.ZodString
+            : T extends SzNumber
+              ? z.ZodNumber
+              : T extends SzBoolean
+                ? z.ZodBoolean
+                : T extends SzBigInt
+                  ? z.ZodBigInt
+                  : T extends SzNaN
+                    ? z.ZodNaN
+                    : T extends SzDate
+                      ? z.ZodDate
+                      : T extends SzUndefined
+                        ? z.ZodUndefined
+                        : T extends SzNull
+                          ? z.ZodNull
+                          : T extends SzSymbol
+                            ? z.ZodSymbol
+                            : T extends SzAny
+                              ? z.ZodAny
+                              : T extends SzUnknown
```

Comment:
Is that a new version of prettier or why did the formatting change so radically?

Reply:
Yeah, a new version of Prettier...